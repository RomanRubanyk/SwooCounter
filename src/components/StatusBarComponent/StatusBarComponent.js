import React, { useContext } from 'react';
import { StatusBar, View, StyleSheet } from 'react-native';
import { ThemeContext } from '../../providers/ThemeProvider';
import { STATUS_BAR_HEIGHT } from '../../constants/general';

const StatusBarComponent = () => {
  const { isDark, colors } = useContext(ThemeContext);

  const styles = StyleSheet.create({
    statusBar: {
      backgroundColor: colors.layout,
      height: STATUS_BAR_HEIGHT,
    },
  });

  StatusBar.currentHeight = STATUS_BAR_HEIGHT;

  return (
    <View style={styles.statusBar}>
      <StatusBar barStyle={isDark ? 'light-content' : 'dark-content'} />
    </View>
  );
};

export default StatusBarComponent;
