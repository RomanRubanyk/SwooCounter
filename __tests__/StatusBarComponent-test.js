import React from 'react';
import renderer from 'react-test-renderer';
import StatusBarComponent from '../src/components/StatusBarComponent/StatusBarComponent'

test('renders correctly', () => {
  const tree = renderer.create(<StatusBarComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
