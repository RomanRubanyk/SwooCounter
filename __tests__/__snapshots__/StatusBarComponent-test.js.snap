// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(0, 71, 97)",
      "height": 10,
    }
  }
/>
`;
